app/models/test.rb

```diff
@@ -1,13 +1,27 @@
 class Test < ApplicationRecord
+  MAX_LEVEL = Float::INFINITY
+
   belongs_to :category
   belongs_to :author, class_name: 'User', foreign_key: 'author_id'
   has_many :questions, dependent: :nullify
   has_many :tests_users, dependent: :nullify
   has_many :users, through: :tests_users
 
-  def self.order_title_desc(category)
-    joins(:category).where(categories: { title: category })
-                    .order('tests.title DESC')
-                    .pluck('tests.title')
-  end
+  scope :easy, -> { where(level: 0..1) }
+  scope :middle, -> { where(level: 2..4) }
+  scope :difficult, -> { where(level: 5..MAX_LEVEL) }
+  scope :order_title_desc, lambda { |category|
+                             joins(:category).where(categories: { title: category })
+                                             .order('tests.title DESC')
+
+                             validates :title, presence: true, uniqueness: { scope: :level,
```

Comment:
[в документации](https://guides.rubyonrails.org/active_record_validations.html#uniqueness) к хелперу `uniqueness` сказано следующее:
> It does not create a uniqueness constraint in the database, so it may happen that two different database connections create two records with the same value for a column that you intend to be unique. To avoid that, you must create a unique index on that column in your database.

другими словами, чтобы эта была настоящая уникальность (не только на уровне приложения, но и на уровне базы), нужно в базе создать уникальный индекс

про правильную миграцию можно посмотреть к примеру на [СО](https://stackoverflow.com/questions/32927262/create-migration-with-uniqueness-to-a-scope)